packages/components/psammead-content-shift-blocker/src/index.test.jsx

```diff
@@ -0,0 +1,398 @@
+/* eslint-disable no-underscore-dangle */
+
+import React from 'react';
+import { render, act } from '@testing-library/react';
+import styled from 'styled-components';
+import ContentShiftBlocker from '.';
+
+let IOInstance;
+let ROInstance;
+
+beforeEach(() => {
+  IOInstance = null;
+  ROInstance = null;
+
+  delete global.CSS; // delete our mocked CSS.supports API so the scroll adjustment logic is the default behaviour
+
+  global.IntersectionObserver = callback => {
+    // return a mock instance and store it globally so we can fire the callback to simulate an intersection event
+    IOInstance = {
+      observe: jest.fn(),
+      disconnect: jest.fn(),
+      fireIntersectEvent: callback,
+    };
+    return IOInstance;
+  };
+
+  global.ResizeObserver = callback => {
+    // return a mock instance and store it globally so we can fire the callback to simulate a resize event
+    ROInstance = {
+      observe: jest.fn(),
+      disconnect: jest.fn(),
+      fireResizeEvent: callback,
+    };
+    return ROInstance;
+  };
+
+  // mock scrollHeight - not supported in jsdom
+  Object.defineProperty(HTMLElement.prototype, 'scrollHeight', {
+    configurable: true,
+    get() {
+      return this._scrollHeight;
+    },
+    set(val) {
+      this._scrollHeight = val;
+    },
+  });
+
+  global.pageYOffset = 300; // set the scroll Y position to an arbitrary number (slightly scrolled down)
+
+  document.body.scrollHeight = 5000; // set the default height of the document
+});
+
+afterEach(jest.clearAllMocks);
+
+global.scrollTo = jest.fn((x, y) => {
+  global.pageYOffset = y;
+});
+
+const Content = styled.div`
+  height: 100px;
+  width: 100px;
+`;
+
+it('should render children', () => {
+  const { getByText } = render(
+    <ContentShiftBlocker>
+      <Content>Test content</Content>
+    </ContentShiftBlocker>,
+  );
+
+  expect(getByText('Test content')).toBeTruthy();
+});
+
+it('should accept string measurements', () => {
+  const stringSize = '10em';
+  const { container } = render(
+    <ContentShiftBlocker initialWidth={stringSize} initialHeight={stringSize}>
+      <Content />
+    </ContentShiftBlocker>,
+  );
+
+  expect(
+    getComputedStyle(container.firstChild).getPropertyValue('width'),
+  ).toEqual(stringSize);
+
+  expect(
+    getComputedStyle(container.firstChild).getPropertyValue('height'),
+  ).toEqual(stringSize);
+});
+
+it('should not resize when in view', () => {
+  const initialSize = 100;
+  const sizeIncrease = 100;
+  const { container } = render(
+    <ContentShiftBlocker initialWidth={initialSize} initialHeight={initialSize}>
+      <Content />
+    </ContentShiftBlocker>,
+  );
+
+  // check initial size
+  expect(
+    getComputedStyle(container.firstChild).getPropertyValue('width'),
+  ).toEqual(`${initialSize}px`);
+
+  expect(
+    getComputedStyle(container.firstChild).getPropertyValue('height'),
+  ).toEqual(`${initialSize}px`);
```

Comment:
I am a big advocate of AAA (Arrange Act Assert) in tests (I think I mentioned it to you (@jroebu14) in a call at one point?). In this test we:
- Setup the component (Arrange)
- Render it (Act)
- Check its initial size (Assert)
- Simulate an intersection (Act)
- Simulate a resize (Act)
- Check the resize did not occur on the wrapper (Assert)

I'd consider seperating this section into a seperate test for the initial size:
- Setup the component (Arrange)
- Render it (Act)
- Check its initial size (Assert)

then another test for suppressing the resize:
- Setup the component (Arrange)
- Render it (Act)
- Simulate an intersection (Act)
- Simulate a resize (Act)
- Check the resize did not occur on the wrapper (Assert)

The idea of it is to make sure tests only fail because one aspect of a component has broken and not others. Also assertions only check the 'end result' not that the setup (Arrange) has been successful. I hope that make sense, please feel free to disagree but fancy putting this out there being new and being a fan of this practice!

Reply:
Great suggestion and thanks for educating on me this best practice 👍